src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +33,156 @@ ActorInit Dm_Hina_InitVars = {
     (ActorFunc)DmHina_Draw,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F470.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F56C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F5AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F63C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F75C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F9AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Draw.s")
+void DmHina_Init(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->unk180 = 1;
+    this->actionFunc = func_80A1F470;
+    this->unk158 = this->actor.world.pos.y;
+    this->actor.focus.pos.y = this->actor.world.pos.y;
+    this->unk148 = 0.0f;
+    this->unk15C = 1.0f;
+    this->actor.focus.pos.x = this->actor.world.pos.x;
+    this->actor.focus.pos.z = this->actor.world.pos.z;
```

Comment:
```suggestion
    this->unk148 = 0.0f;
    this->unk15C = 1.0f;
    this->actor.focus.pos.x = this->actor.world.pos.x;
    this->actor.focus.pos.y = this->actor.world.pos.y;
    this->actor.focus.pos.z = this->actor.world.pos.z;
```
Nit: Ordering of x,y,z coords.